### PR TITLE
Libraries should implement our exceptions

### LINKED ISSUE #20: Is it a requirement to implement NetworkException and RequestException?
Are we okey with an implementing library that does not implement `NetworkException` and `RequestException`? Could you really be a HTTP client without those exceptions?

On one hand, what kind of a poor client are you building if you do not support those exceptions...
But on the other hand, the only effect would be that the consuming library does not have a change to recover from the exception.

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #20.

### CODE REVIEW

File: proposed/http-client/http-client.md [hunk 1]
```diff
@@ -27,6 +27,10 @@ should not follow redirect nor throw exceptions on HTTP responses with status 4x
 The following interfaces MAY be implemented together within a single class or
 in separate classes.
 
+An implementing library MUST implement `Psr\Http\Client\Exception` for each exception it throws. 
+It SHOULD implement exceptions for `Psr\Http\Client\Exception\NetworkException` and
+`Psr\Http\Client\Exception\RequestException`.
```
Comment (dbu) [inline proposed/http-client/http-client.md]: should we give an explanation why it should do that? to allow for the client to more precisely act on errors?

for the flow of this document: the paragraph above says "The following interfaces..." and now we add another one talking about exceptions. could we move this somethwere else? maybe a chapter on exceptions?

**Comment by Nyholm** [comment]
I agree. It was misplaced. But I do not think it should be next to the exception interfaces. I've moved it up to the specification.

**Comment by dbu** [comment]
yep, that feels a lot better. this is good to merge.

**Comment by Nyholm** [comment]
Thank you.

**Pull request merged**